src/main/java/au/org/emii/ncdfgenerator/AttributeValueParser.java

```diff
@@ -0,0 +1,202 @@
+{
+	// follows ncdf string conventions
+
+	private int skipWhite(String s, int pos)
+	{
+		while( Character.isSpaceChar(peekChar(s , pos) ))
+			++pos;
+		return pos;
+	}
+
+	private int peekChar(String s, int pos)
+	{
+		if(pos < s.length() )
+			return s.charAt( pos );
+		return -1;
+	}
+
+	public AttributeValue parse( String s ) throws NcdfGeneratorException
+	{
+		int pos = 0;
+
+		AttributeValue a = parseAttributes( s, pos ); 
+		if( a != null) {
+			// ensure we got to the end, with nothing trailing
+			pos = a.pos;
+			pos = skipWhite( s, pos);
+			if( pos == s.length() )
+				return a;
+		}
+
+		throw new NcdfGeneratorException( "Couldn't parse attribute value '" + s + "'" );
+	}
+
+	
+	protected AttributeValue parseAttributes( String s, int pos ) throws NcdfGeneratorException
+	{
+		// try to parse as an attribute array, fallback to creating a scalar 
+
+		List<AttributeValue> items = new ArrayList<AttributeValue> (); 
+		AttributeValue a = null;
+		do { 
+			a = parseAttributeValue1( s, pos ); 
+			if( a != null) { 
+				items.add( a) ; 
+				pos = a.pos;
+				pos = skipWhite( s, pos ); 
+			}
+		}
+		while( a != null ); 
+
+		if( items.size() == 1 ) { 
+			return items.get( 0); 
+		}
+		else if ( items.size() > 1 ) {
+			// handle single dimension only at this point...
+			int shape [] = { items.size() };
+
+			// assume type according to the first value
+			Object first = items.get( 0).value;
+			Array ar = null;
+
+			if( first instanceof Byte ) { 
+				ar = Array.factory( DataType.BYTE, shape );
+				int i = 0; 
+				for( AttributeValue e : items )
+					ar.setByte( i++, (Byte)e.value );
+			}
+			else if( first instanceof Integer ) { 
+				ar = Array.factory( DataType.INT, shape );
+				int i = 0; 
+				for( AttributeValue e : items )
+					ar.setInt( i++, (Integer)e.value );
+			}
+			// TODO long
+			else if( first instanceof Float ) { 
+				ar = Array.factory( DataType.FLOAT, shape );
+				int i = 0; 
+				for( AttributeValue e : items )
+					ar.setFloat( i++, (Float)e.value );
+			}
+			else if( first instanceof Double ) { 
+				ar = Array.factory( DataType.DOUBLE, shape );
+				int i = 0; 
+				for( AttributeValue e : items )
+					ar.setDouble( i++, (Double)e.value );
+			}
+			// don't think strings are supported by api?
+			else {
+				// more a runtime exception
+				throw new NcdfGeneratorException( "Unknonwn array value type '" + first.getClass().getName() + "'");
+			}
+			return new AttributeValue(pos, ar );
+		}
+		else
+			return null;
+
+	}
+	
+
+	protected AttributeValue parseAttributeValue1( String s, int pos )
+	{
+		AttributeValue a = parseFloat( s, pos );
+		if( a != null )
+			return a;
+
+		a = parseInteger( s, pos );
+		if( a != null )
+			return a;
+
+		a = parseString( s, pos );
+		if( a != null)
+			return a;
+		return null;
+	}
+
+	protected AttributeValue parseFloat( String s, int pos )
+	{
+		int pos2 = pos;
+
+		if( peekChar( s, pos2) == '-')
+			++pos2;
+
+		boolean gotDot = false;
```

Comment:
My understanding is that java convention favors placing declarations close to their usage (like c++/c# and unlike C). If you have a reference that says otherwise I'll change it.